GDJS/GDJS/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -1096,33 +1076,35 @@ gd::String EventsCodeGenerator::GenerateConditionsListCode(
     gd::EventsCodeGenerationContext& context) {
   gd::String outputCode;
 
-  for (std::size_t i = 0; i < conditions.size(); ++i)
     outputCode += GenerateBooleanInitializationToFalse(
-        "condition" + gd::String::From(i) + "IsTrue", context);
+        "isConditionTrue", context);
 
   for (std::size_t cId = 0; cId < conditions.size(); ++cId) {
-    if (cId != 0)
-      outputCode +=
-          "if ( " +
-          GenerateBooleanFullName(
-              "condition" + gd::String::From(cId - 1) + "IsTrue", context) +
-          ".val ) {\n";
-
     gd::String conditionCode =
         GenerateConditionCode(conditions[cId],
-                              "condition" + gd::String::From(cId) + "IsTrue",
+                              "isConditionTrue",
                               context);
     if (!conditions[cId].GetType().empty()) {
+      for (std::size_t i = 0; i < cId;
+           ++i)  // Skip conditions if one condition is false. //TODO : Can be
+                 // optimized
+      {
+        if (i == 0)
+          outputCode += "if (";
+        else
+          outputCode += " && ";
+        outputCode += GenerateBooleanFullName("isConditionTrue", context);
+        if (i == cId - 1) outputCode += ") ";
```

Comment:
Are you sure about this? Is this not generating `if (isConditionTrue && isConditionTrue && isConditionTrue)`?

Reply:
(with the "depth" of the condition in the name of the boolean, but this is always the same here anyway?)

Reply:
Yes, I copied it from the C++ generator and forget to come back to it. I'll remove the for loop.

The generated code look like this:
```
let isConditionTrue_0 = false;
      {
      isConditionTrue_0 = false;
      {isConditionTrue_0 = (1 == 1);
      }
      }
      if (isConditionTrue_0) {
      isConditionTrue_0 = false;
      {isConditionTrue_0 = (1 == 1);
      }
      }
      if (isConditionTrue_0) {
      {runtimeScene.getScene().getVariables().get("SuccessVariable").setNumber(1);
      }}

      }
```